thorntail-runner/src/main/java/org/wildfly/swarm/runner/RunnerConstants.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2015-2017 Red Hat, Inc, and individual contributors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.swarm.runner;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * @author Michal Szynkiewicz, michal.l.szynkiewicz@gmail.com
+ * <br>
+ * Date: 9/11/18
+ */
+public class RunnerConstants {
+    public static final String CACHE_STORAGE_DIR = loadProperty("thorntail.runner.cache-location", ".thorntail-runner-cache");
+    public static final String IGNORE_DEFAULT_REPOSITORIES = loadProperty("thorntail.runner.ignore.default.repositories", null);
+    public static final String USER_REPOSITORIES = loadProperty("thorntail.runner.repositories", null);
+    public static final String PRESERVE_JAR = loadProperty("thorntail.runner.preserve-jar", null);
+
+    private RunnerConstants() {
+    }
+
+    private static String loadProperty(String propertyName, String defaultValue) {
+        String propertyValue;
+        try (InputStream st = new FileInputStream("thorntail-runner.properties")) {
```

Comment:
This will unnecessarily load the file for each property, again and again. How about creating a `static` block that will initialize all the constants at once?

Reply:
Adapted code, I was a bit twisted in just creating a static block.. I rather chose to rework it to to something cleaner (where constants and configurations logic are nicely seperated). Let me know what you think